recipes/log4cxx/all/conanfile.py

```diff
@@ -61,33 +86,57 @@ def layout(self):
         self.requires("apr-util/1.6.1")
-        self.requires("expat/2.5.0")
-        if self.settings.os != "Windows":
+        if self.options.get_safe("with_odbc_appender") and self.settings.os != "Windows":
             self.requires("odbc/2.3.11")
+        if self.options.get_safe("with_smtp_appender"):
+            self.requires("libesmtp/1.1.0")
+        if self.options.get_safe("with_fmt_layout"):
+            self.requires("fmt/9.1.0")
+        if self.options.get_safe("with_qt"):
+            self.requires("qt/5.15")
 
     def validate(self):
-        # TODO: if compiler doesn't support C++17, boost can be used instead
-        if self.settings.compiler.get_safe("cppstd"):
-            check_min_cppstd(self, self._min_cppstd)
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(f"{self.ref} requires a compiler that supports at least C++{self._min_cppstd}")
+        if Version(self.version) < "1.0.0" or self.options.get_safe("with_multiprocess_rolling_file_appender"):
+            # TODO: if compiler doesn't support C++17, boost can be used instead
+            if Version(self.version) < "1.0.0":
+                self.output.info(f"Version {self.version} requires C++17. log4cxx version 1.1.0 does not.")
+            else:
+                self.output.info("multiprocess rolling file appender requires C++17.")
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            compiler_version = Version(self.settings.compiler.version)
+            if not minimum_version:
+                self.output.warning("Your compiler is unknown. Assuming it supports C++17.")
+            elif compiler_version < minimum_version:
+                raise ConanInvalidConfiguration(f"{self.settings.compiler} {compiler_version} does not support C++17: {minimum_version} required.")
+            if self.settings.compiler.get_safe("cppstd"):
+                check_min_cppstd(self, "17")
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
 
     def build_requirements(self):
         if self.settings.os != "Windows":
             if not self.conf.get("tools.gnu:pkg_config", check_type=str):
                 self.tool_requires("pkgconf/2.0.3")
 
     def source(self):
-        # OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect:
-        # 'src\\test\\resources\\output\\xyz\\:'
-        pattern = "*[!:]"
-        get(self, **self.conan_data["sources"][self.version], strip_root=True, pattern=pattern)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
         tc.cache_variables["BUILD_TESTING"] = False
         tc.cache_variables["LOG4CXX_INSTALL_PDB"] = False
```

Comment:
```suggestion
        tc.variables["BUILD_TESTING"] = False
        tc.variables["LOG4CXX_INSTALL_PDB"] = False
```
log4cxx/0.12 sets CMake policy version to 3.13, so `tc.variables` can be used safely.